src/codeu/chat/DerbyStore.java

```diff
@@ -356,6 +367,45 @@ public Message getMessage(Uuid messageid) throws SQLException, IOException {
 		return allMessages;
 	}
 	
+	@Override
+	public int getMessageCount(Uuid userid) throws SQLException {
+		PreparedStatement getMessageCount = conn.prepareStatement(getMessageCountById);
+		getMessageCount.setString(1, removeCharsInUuid(userid.toString()));
+		
+		ResultSet messageCount = getMessageCount.executeQuery();
+		
+		if (messageCount.next()) {
+			return messageCount.getInt(DBColumns.UserMessageCountTable.messageCount);
+		}
+		
+	return 0;
+	}
+
+	@Override
+	public int setUserMessageCount(Uuid userid) throws SQLException {
```

Comment:
The function seems to be doing a lot. Isn't it really incrementing the message count? As always: comments would help.